src/Agent.Listener/DistributedTask.Pipelines/Yaml/Contracts/CheckoutStep.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.TypeConverters;
+
+namespace Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.Contracts
+{
+    internal sealed class CheckoutStep : ISimpleStep
+    {
+        public String Name { get; set; }
+
+        internal String Clean { get; set; }
```

Comment:
No the CheckoutStep properties are task inputs and are sent down to the agent, and will be handled by the agent's task infrastructure and processed like all other task inputs (future plan, temporary hack today and are converted to job variables).

For properties that get processed on the server side (not sent down to agent for evaluation), the technique you are mentioning will be used in some places. However, the types in the contracts I am editing here, will be all strings because they are internal contracts for deserialization only. The deserialized result will be converted to the runtime contracts on the server side when a build is queued. YAML isn't the only entry point. We also have the existing web definitions. Both will convert into the runtime types. The `YamlValue<Boolean>` is a server-side runtime concept.

I'm curious why the interest in the details :) Are you planning to use local run? Or just curious where the yaml schema is landing? Is there a scenario you are interested in, that you are trying to understand whether we are planning for (feedback welcome)?